godotenv.go

```diff
@@ -39,18 +41,35 @@ const doubleQuoteSpecialChars = "\\\n\r\"!$`"
 //		godotenv.Load("fileone", "filetwo")
 //
 // It's important to note that it WILL NOT OVERRIDE an env variable that already exists - consider the .env file to set dev vars or sensible defaults
-func Load(filenames ...string) (err error) {
+func LoadWithLookupFn(lookupFn func(string) (string, bool), filenames ...string) (err error) {
 	filenames = filenamesOrDefault(filenames)
 
 	for _, filename := range filenames {
-		err = loadFile(filename, false)
+		err = loadFile(filename, false, lookupFn)
 		if err != nil {
 			return // return early on a spazout
 		}
 	}
 	return
 }
 
+// Load will read your env file(s) and load them into ENV for this process.
+//
+// Call this function as close as possible to the start of your program (ideally in main)
+//
+// If you call Load without any args it will default to loading .env in the current path
+//
+// You can otherwise tell it which files to load (there can be more than one) like
+//
+//		godotenv.Load("fileone", "filetwo")
+//
+// It's important to note that it WILL NOT OVERRIDE an env variable that already exists - consider the .env file to set dev vars or sensible defaults
+func Load(filenames ...string) (err error) {
+	return LoadWithLookupFn(func(string) (string, bool){
```

Comment:
Should this one have used `noLookupFn` as well? 

But perhaps instead of a dummy function, pass `nil` as lookup-function, then where it's executed, check if it's nil and if so, skip executing?